src/main.c

```diff
@@ -72,12 +71,12 @@ void run_simulation()
 
 
     /* Init population */
-    sc_population population;
-    population_create(population_size, &population, &sys, &goal);
+    sc_population *population=(sc_population *)malloc(sizeof(sc_population));
+    population_create(sys.no_of_programs, population, &sys, &goal);
```

Comment:
This will work, but I don't think there's a need to dynamically allocate the population here.